### PR TITLE
Fade in fade out when switching websites [fix for #47, #9]

### PR DESCRIPTION
**Pull request by uruha-komachin**
Using another SSWebView, `webview2` in WebViewController to fix issue #47 and #9.

Animation time is set to 0.5 seconds for both fade in and fade out effect. I'm not sure if the animation time brings the best user experience, but please advise and I can make corresponding changes.

Btw, amazing application! 😉

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for working on this. I think the code needs to be better abstracted. Having to handle either web view in every single place is not going to be sustainable.

**Comment by uruha-komachin** [comment]
@sindresorhus Hi, I've tried to abstract a little bit more on switching the background webView and the foreground one. Hope that's more clear and maintainable now.

**Comment by sindresorhus** [comment]
The PR doesn't compile.

File: Plash/WebViewController.swift [hunk 1]
```diff
@@ -145,15 +157,15 @@ extension WebViewController: WKNavigationDelegate {
...
 	}
 
 	func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
-		internalOnLoaded(error)
+		internalOnLoaded(error, webView as! SSWebView)
 	}
 
 	func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
-		internalOnLoaded(error)
+		internalOnLoaded(error, webView as! SSWebView)
```
Comment (sindresorhus) [inline Plash/WebViewController.swift:168]: Don't force-cast.

File: Plash/WebViewController.swift [hunk 2]
```diff
@@ -72,40 +72,52 @@ final class WebViewController: NSViewController {
...
 	func recreateWebView() {
-		webView = createWebView()
-		view = webView
+		backgroundWebView = createWebView()
+		backgroundWebView.isHidden = true
 	}
 
 	lazy var webView = createWebView()
+	lazy var backgroundWebView = createWebView()
+	var loadInBackground = false
```
Comment (sindresorhus) [inline Plash/WebViewController.swift:81]:
```suggestion
	var isLoadedInBackground = false
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
I think this PR generally needs more work. Try to refactor it a bit more. Also make sure you try out a lot of different combination of use-cases, to route out bugs.

**Comment by sindresorhus** [comment]
Closing for lack of activity and it's too far from being mergeable.

**Pull request closed without merging**